patroni/postgresql.py

```diff
@@ -1294,7 +1294,9 @@ def pick_synchronous_standby(self, cluster):
                      FROM pg_stat_replication
                     ORDER BY flush_location DESC"""):
             member = members.get(app_name)
-            if state != 'streaming' or not member or member.tags.get('nosync', False):
+            if (state != 'streaming' or not member
+                    or member.tags.get('nosync', False)
+                    or member.tags.get('nofailover', False)):
```

Comment:
Just a nitpicking: `Member` object has `nofailover` attribute, so there is no need to get it from tags.
Line could be written as:
```python
if state != 'streaming' or not member or member.nofailover or member.tags.get('nosync', False):
```

Also I'm wondering, is it really necessary to do this? At the end we have a `nosync` tag, which does exactly what we want. Replica wont became a synchronous if `nosync` is set in `patroni.yaml`.

Also by introducing this additional check you are limiting some future functionality. Starting from 9.6 it's possible to have more than 1 synchronous standby (although it's not yet supported by Patroni). So in theory, by playing with nosync and nofailover flags it would be possible to have a cluster let's say with 2 synchronous replicas, but one of them will have a `nofailver` flag set, just because we want to have it always not only read-only, but also synchronous.